src/transformers/models/auto/modeling_auto.py

```diff
@@ -147,7 +147,7 @@
 MODEL_WITH_LM_HEAD_MAPPING_NAMES = OrderedDict(
     [
         # Model with LM heads mapping
-        ("imagegpt", "ImageGPTForCausalLM"),
+        ("imagegpt", "ImageGPTForCausalImageModeling"),
```

Comment:
As you can see, ImageGPT is currently included in the MODEL_WITH_LM_HEAD_MAPPING_NAMES.

Should we create a MODEL_WITH_IMAGE_MODELING_HEAD_MAPPING_NAMES instead, or this is ok?

Reply:
I would put it in a separate mapping indeed!

Reply:
I would remove it from that mapping that is deprecated anyway, and just use another mapping `MODEL_FOR_CAUSAL_IMAGE_MODELING_MAPPING` (and MAPPING_NAMES)